firebase-firestore/src/test/java/com/google/firebase/firestore/FirestoreMultiDbComponentTest.java

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
+
+import android.content.Context;
+
+import androidx.test.platform.app.InstrumentationRegistry;
+
+import com.google.firebase.FirebaseApp;
+import com.google.firebase.FirebaseOptions;
+import com.google.firebase.auth.internal.InternalAuthProvider;
+import com.google.firebase.firestore.remote.GrpcMetadataProvider;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.junit.Assert.assertNotSame;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class FirestoreMultiDbComponentTest {
+
+  @Test
+  public void onDeleted_shouldRemoveAllInstances() {
+    FirebaseApp firebaseApp = createApp("onDeleted-shouldRemoveAllInstances");
+    FirestoreMultiDbComponent component = createComponent(firebaseApp);
+    FirebaseFirestore firebaseFirestore1 = component.get("db1");
+    FirebaseFirestore firebaseFirestore2 = component.get("db2");
+
+    component.onDeleted(firebaseApp.getName(), firebaseApp.getOptions());
+
+    assertNotSame(firebaseFirestore1, component.get("db1"));
+    assertNotSame(firebaseFirestore2, component.get("db2"));
+    // Clean up the instances registered by the calls above.
+    component.onDeleted(firebaseApp.getName(), firebaseApp.getOptions());
```

Comment:
Is this cleanup actually required? There's no shared state, so it seems like it would be fine to just drop the component and delete this line and the comment.

If this clean-up *is* required, then this should probably be within a finally block, because `assertNotSame` will throw on a failure and then the cleanup won't happen.

Reply:
No, it's probably not required. I've removed it.